Core/Core/Planner/CalendarToDo/ViewModel/EditCalendarToDoViewModel.swift

```diff
@@ -49,46 +85,79 @@ final class EditCalendarToDoViewModel: ObservableObject {
     // MARK: - Input
 
     let didTapCancel = PassthroughSubject<Void, Never>()
-    let didTapAdd = PassthroughSubject<Void, Never>()
+    let didTapSave = PassthroughSubject<Void, Never>()
 
     // MARK: - Private
 
+    private let mode: Mode
     private let toDoInteractor: CalendarToDoInteractor
     private let calendarListProviderInteractor: CalendarFilterInteractor
-    private var subscriptions = Set<AnyCancellable>()
     private var selectedCalendar = CurrentValueSubject<CDCalendarFilterEntry?, Never>(nil)
+    private var isTouched: Bool = false
+
+    private var subscriptions = Set<AnyCancellable>()
 
     // MARK: - Init
 
     init(
+        plannable: Plannable? = nil,
         toDoInteractor: CalendarToDoInteractor,
         calendarListProviderInteractor: CalendarFilterInteractor,
         completion: @escaping (PlannerAssembly.Completion) -> Void
     ) {
         self.toDoInteractor = toDoInteractor
         self.calendarListProviderInteractor = calendarListProviderInteractor
 
-        // end of today, to match default web behaviour
-        date = .now.endOfDay()
+        if let plannable {
```

Comment:
You could create a local `weak var weakSelf = self` property above this line so you wouldn't need to capture self in every closure.

Reply:
I simplified these a bit, but I would rather keep the separate capturings.
My main reason is syntax highlight, which shows `self?` as a keyword, opposed to `weakSelf` which is colored as a local variable (i.e.: it's not colored).
Another reason is that I feel it's more explicit this way.
But if you guys insist I can change it.